semgrep/semgrep/evaluation.py

```diff
@@ -373,9 +373,27 @@ def evaluate(
         logger.debug(f"compiled result {valid_ranges_to_output}")
         logger.debug("-" * 80)
 
+    # Addresses https://github.com/returntocorp/semgrep/issues/1699,
+    # where metavariables from pattern-inside are not bound to messages.
+    # This should handle cases with pattern + pattern-inside. This doesn't handle
+    # pattern-not-inside because it is difficult to determine metavariables for
+    # exclusion ranges. For example: imaagine a pattern-not-inside for 'def $CLASS(): ...'
```

Comment:
```suggestion
    # exclusion ranges. For example: imagine a pattern-not-inside for 'def $CLASS(): ...'
```

Reply:
IMO these should go in the ticket for metavariables in pattern-not-inside instead of as a comment in the code